src/stream/src/executor/flow_control.rs

```diff
@@ -32,23 +32,40 @@ use super::*;
 /// It is used to throttle problematic MVs that are consuming too much resources.
 pub struct FlowControlExecutor {
     input: BoxedExecutor,
+    actor_ctx: ActorContextRef,
     rate_limit: Option<u32>,
 }
 
 impl FlowControlExecutor {
     #[allow(clippy::too_many_arguments)]
```

Comment:
```suggestion
```

Reply:
> A bit ugly here. Why not just add an iter for it?

I rewrite this part and I think the original version is more reader-friendly.

```rust
        let table_id_to_apply: HashSet<TableId> =
            map.iter()
                .filter_map(|(table_id, table_fragments)| {
                    if table_fragments.fragments.values().any(|stream_node| {
                        stream_node.actors.iter().any(|actor| {
                            if let Some(stream_node) = actor.nodes.as_ref() {
                                if let Some(NodeBody::Source(ref node)) = stream_node.node_body.as_ref() {
                                    if let Some(node_inner) = &node.source_inner && node_inner.source_id == source_id as u32 {
                                        true
                                    } else { false }
                                } else {
                                    false
                                }

                            } else { false }
                        })
                    }) {
                        Some(*table_id)
                    } else {
                        None
                    }
                })
                .collect();
```